client/models/scenery_editor.js

```diff
@@ -318,16 +321,71 @@ class SceneryEditor {
             post('api/tools/scenery/instance/delete/' + WORKSPACES.opened, {
                 id: document.getElementById('tools-detail-scenery-id').innerText
             }, () => {
-                WORKSPACES.reload();
+                this.removeObject(document.getElementById('tools-detail-scenery-id').innerText);
             });
         } else if (this.selected_type === 'unique') {
             post('api/tools/scenery/unique/delete/' + WORKSPACES.opened, {
                 id: document.getElementById('tools-detail-scenery-id').innerText
             }, () => {
-                WORKSPACES.reload();
+                this.removeObject(document.getElementById('tools-detail-scenery-id').innerText);
             });
         }
     }
+
+    addObject(object){
+        let object_id = object.x + ',' + object.y;
+            
+        WORKSPACES.current_map.sceneryLoader.createScenery(object, (model, definition) => {
+            let m = createSceneryMesh(object_id, object, WORKSPACES.current_map.terrain, model, definition);
+            m.original_id = { type: 'scenery', id: object_id };
+            WORKSPACES.current_map.scenery_groups['trees'].add(m);
+            WORKSPACES.current_map.scenery_references[object_id] = {
+                instance: object,
+                definition: definition,
+                threeObject: m
+            }
+        })
+    }
+
+    editObject(object){
+        let newObject = WORKSPACES.current_map.scenery_references[object.id].instance;
+
+        if (object.modification == 'tint'){
+            this.removeObject(object.id);
+            if ('tint' in object){
+                newObject.tint = object.tint;
+            } else{
+                newObject.tint = object.remove_tint;
+            }
+            this.addObject(newObject);
+        }
+
+        if (object.modification == 'rotation'){
+            this.removeObject(object.id);
+            newObject.rotation = object.rotation;
+            this.addObject(newObject);
+        }
+    }
+
+    removeObject(object){
+        if (object.object == "delete") object = object.x + ',' + object.y;
```

Comment:
What is this abomination of code?

Reply:
Don't ever overwrite references like this, it makes them unreadable, as you can see by the next comment.